bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileAdvisor.java

```diff
@@ -41,4 +42,20 @@
     @Nullable
     ProfileTypeUID getSuggestedProfileTypeUID(Channel channel, @Nullable String itemType);
 
+    /**
+     * Suggest a custom profile for a given {@link ChannelType} (and potentially also the itemType).
+     *
+     * Please note:
+     * <ul>
+     * <li>This will override any default behavior
+     * <li>A "profile" configuration on the link will override this suggestion
+     * </ul>
+     *
+     * @param channelType the {@link ChannelType} of the linked channel
+     * @param itemType the linked itemType (not applicable for trigger channels)
```

Comment:
I thought trigger channels can now be linked to items, can't they?

Reply:
Yes, they do. The suggested ProfileTypeUID however will not depend on the given itemType when it comes to trigger channels.